### PR TITLE
vimPlugins.hare-vim: init at 2022-07-02

### PR DESCRIPTION
**Pull request by WhyNotHugo**
###### Description of changes

Added a new vim plugin: hare-vim

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/vim/plugins/generated.nix [hunk 1]
```diff
@@ -3346,6 +3346,17 @@ final: prev:
...
   };
 
+  hare-vim = buildVimPluginFrom2Nix {
+    pname = "hare.vim";
+    version = "2022-07-02";
+    src = fetchgit {
+            url = "https://git.sr.ht/~sircmpwn/hare.vim";
+            rev = "5c758cdbbabd6e4ba92bced9428cd1fa4212f003";
+            sha256 = "0l1q9x1n30h32zfnd2krg3frywydfmb8ic05619pylamnyh4w918";
+        };
```
Comment (azahi) [inline pkgs/applications/editors/vim/plugins/generated.nix:3356]: Please use [`fetchFromSourceHut`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/fetchsourcehut/default.nix) and make sure you indent it out with 2 spaces.

**Review by azahi — CHANGES_REQUESTED** [review]

File: pkgs/applications/editors/vim/plugins/generated.nix (hunk 1 above)
Comment (WhyNotHugo) [reply pkgs/applications/editors/vim/plugins/generated.nix]: Fixed. Not sure how I messed up indentation; I'd used the `update.py` script to generate this.

File: pkgs/applications/editors/vim/plugins/generated.nix [hunk 2]
```diff
@@ -3346,6 +3346,17 @@ final: prev:
...
   };
 
+  hare-vim = buildVimPluginFrom2Nix {
+    pname = "hare.vim";
+    version = "2022-07-02";
+    src = fetchFromSourceHut {
+      url = "https://git.sr.ht/~sircmpwn/hare.vim";
+      rev = "5c758cdbbabd6e4ba92bced9428cd1fa4212f003";
+      sha256 = "0l1q9x1n30h32zfnd2krg3frywydfmb8ic05619pylamnyh4w918";
+    };
```
Comment (azahi) [inline pkgs/applications/editors/vim/plugins/generated.nix:3356]:
```suggestion
    src = fetchFromSourcehut {
      owner = "~sircmpwn";
      repo = "hare.vim";
      rev = "5c758cdbbabd6e4ba92bced9428cd1fa4212f003";
      sha256 = "0l1q9x1n30h32zfnd2krg3frywydfmb8ic05619pylamnyh4w918";
    };
```

**Review by azahi — CHANGES_REQUESTED** [review]

File: pkgs/applications/editors/vim/plugins/generated.nix (hunk 2 above)
Comment (WhyNotHugo) [reply pkgs/applications/editors/vim/plugins/generated.nix]: Right! Not sure what I did testing this that it worked :sweat: 

Updated correctly now.

File: pkgs/applications/editors/vim/plugins/generated.nix [hunk 3]
```diff
@@ -3346,6 +3346,18 @@ final: prev:
     meta.homepage = "https://github.com/junegunn/gv.vim/";
   };
 
+  hare-vim = buildVimPluginFrom2Nix {
+    pname = "hare.vim";
+    version = "2022-07-02";
+    src = fetchFromSourceHut {
```
Comment (azahi) [inline pkgs/applications/editors/vim/plugins/generated.nix:3352]: Almost there, you've missed this part :)
```suggestion
    src = fetchFromSourcehut {
```
My initial suggestion had the _H_ capitalized, sorry.

**Review by azahi — CHANGES_REQUESTED** [review]

File: pkgs/applications/editors/vim/plugins/generated.nix (hunk 3 above)
Comment (WhyNotHugo) [reply pkgs/applications/editors/vim/plugins/generated.nix]: Ah, no wonder I found no example usages!

**Comment by azahi** [comment]
Apologies, I didn't realize that this file is autogenerated. The issue now is that `update.py` doesn't generate a `fetchFromSourcehut` clause, so it uses `fetchgit` for non-GitHub repos. I've tested generation out and it seems like `fetchgit` part doesn't indent correctly while being autogenerated, here's an [example](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/vim/plugins/generated.nix#L4133-L4137) from the `master` branch. 

This issue needs to be fixed but that gets out of the scope of this PR. Please revert to your original [commit](https://github.com/NixOS/nixpkgs/commit/bfb4f73e5572f22e77e0e9c74f8be2c7b141f7fd) and sorry for the noise again :(

**Comment by WhyNotHugo** [comment]
Oh, right, next update would undo the indentation and add lots of noise.

Reverted.

**Review by azahi — APPROVED** [review]

**Comment by WhyNotHugo** [comment]
@teto Perhaps you're the right person for review here?

File: pkgs/applications/editors/vim/plugins/generated.nix (hunk 1 above)
Comment (teto) [reply pkgs/applications/editors/vim/plugins/generated.nix:3356]: It's probably the script that doesn't indent properly. To use [fetchFromSourcehut](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/fetchsourcehut/default.nix), one would need to change the updater  script too.

**Comment by teto** [comment]
> Apologies, I didn't realize that this file is autogenerated. The issue now is that update.py doesn't generate a fetchFromSourcehut clause, so it uses fetchgit for non-GitHub repos. I've tested generation out and it seems like fetchgit part doesn't indent correctly while being autogenerated, here's an [example](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/vim/plugins/generated.nix?rgh-link-date=2022-07-25T12%3A34%3A00Z#L4133-L4137) from the master branch.

Anyone welcome to fix that. It doesnt block this PR though

File: pkgs/applications/editors/vim/plugins/generated.nix (hunk 1 above)
Comment (WhyNotHugo) [reply pkgs/applications/editors/vim/plugins/generated.nix:3356]: Honestly, I prefer seeing the URL here. It's easy to copy-paste to find upstream. Using `fetchFromSourceHut` obfuscates the URL a lot.

**Pull request merged**